unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/WebRequestController.cs

```diff
@@ -146,7 +146,7 @@ public static WebRequestController Create()
                 {
                     if (resultOp.webRequest.WebRequestSucceded())
                     {
-                        OnSuccess?.Invoke(resultOp.webRequest);
+                        OnSuccess?.Invoke(resultOp);
```

Comment:
Why do tou need to return the custom async operation and not directly the unity webRequest as before?

Reply:
This is a necessary step in order to move the WebRequestAsync operation to an interface, since the webrequest is a class from unity it can't derive from an interface

Reply:
I don't understand... the class that we are moving to an interface is `WebRequestAsyncOperation` and not `UnityWebRequest`. The fact of `WebRequestAsyncOperation` is using methods or properties of type `UnityWebRequest` shouldn't be a problem, no?

Reply:
Yes it is. If you want to mock up the response of the the GET operation, you will need the ability to substitute the WebRequestAsyncOperation, if you are using UnityWebRequest you cant mock up it since you can make it an interface. 

Example of code:
![image](https://user-images.githubusercontent.com/54064092/141078411-a1bad726-3543-4273-ac55-8946adb1883a.png)

Reply:
Ok, you say that, in your example, you couldn't replace this:
![image](https://user-images.githubusercontent.com/64659061/141089288-01758f2c-b9bc-4e93-8cee-70b7476d6fe9.png)
By: `x.ArgAt<Action<UnityWebRequest>>(sucessParamIndex).Invoke(new UnityWebRequest())`

Right?

Reply:
If you do that the thing is that you can't get the data that you want from the `new UnityWebRequest()`. 

In this example, you can see that the Substitue for the `IWebRequestAsyncOperation` is able to return the Json data because i am able to configure it to return the json.

Since you can't configure the UnityWebRequest to return the json, you won't be able to mockup the complete system to simulate an https call and return the data from